src/Elastic.CommonSchema.NLog/EcsLayout.cs

```diff
@@ -0,0 +1,432 @@
+			var processTitle = ProcessTitle?.Render(logEventInfo);
+			var processExecutable = ProcessExecutable?.Render(logEventInfo);
+			var processThreadId = ProcessThreadId?.Render(logEventInfo);
+
+			if (string.IsNullOrEmpty(processId)
+				&& string.IsNullOrEmpty(processName)
+				&& string.IsNullOrEmpty(processTitle)
+				&& string.IsNullOrEmpty(processExecutable)
+				&& string.IsNullOrEmpty(processThreadId))
+				return null;
+
+			return new Process
+			{
+				Title = processTitle,
+				Name = processName,
+				Pid = !string.IsNullOrEmpty(processId) ? long.Parse(processId) : 0,
+				Executable = processExecutable,
+				Thread = !string.IsNullOrEmpty(processId) ? new ProcessThread { Id = long.Parse(processId) } : null
+			};
+		}
+
+		private Server GetServer(LogEventInfo logEventInfo)
+		{
+			var serverUser = ServerUser?.Render(logEventInfo);
+			var serverAddress = ServerAddress?.Render(logEventInfo);
+			var serverIp = ServerIp?.Render(logEventInfo);
+			if (string.IsNullOrEmpty(serverUser) && string.IsNullOrEmpty(serverAddress) && string.IsNullOrEmpty(serverIp))
+				return null;
+
+			return new Server
+			{
+				User = !string.IsNullOrEmpty(serverUser)
+					? new User { Name = serverUser }
+					: null,
+				Address = serverAddress,
+				Ip = serverIp
+			};
+		}
+
+		private Trace GetTrace(LogEventInfo logEventInfo)
+		{
+			var traceId = ApmTraceId?.Render(logEventInfo);
+			if (string.IsNullOrEmpty(traceId))
+				return null;
+
+			return new Trace
+			{
+				Id = traceId
+			};
+		}
+
+		private Transaction GetTransaction(LogEventInfo logEventInfo)
+		{
+			var transactionId = ApmTransactionId?.Render(logEventInfo);
+			if (string.IsNullOrEmpty(transactionId))
+				return null;
+
+			return new Transaction
+			{
+				Id = transactionId
+			};
+		}
+
+		private Host GetHost(LogEventInfo logEventInfo)
+		{
+			var hostId = HostId?.Render(logEventInfo);
+			var hostName = HostName?.Render(logEventInfo);
+			var hostIp = HostIp?.Render(logEventInfo);
+
+			if (string.IsNullOrEmpty(hostId)
+				&& string.IsNullOrEmpty(hostName)
+				&& string.IsNullOrEmpty(hostIp))
+				return null;
+
+			var host = new Host
+			{
+				Id = hostId,
+				Name = hostName,
+				Ip = string.IsNullOrEmpty(hostIp) ? null : new[] { hostIp }
+			};
+
+			return host;
+		}
+
+		private static long GetSysLogSeverity(LogLevel logLevel)
+		{
+			if (logLevel == LogLevel.Trace || logLevel == LogLevel.Debug)
+				return 7;
+
+			if (logLevel == LogLevel.Info)
+				return 6;
+
+			if (logLevel == LogLevel.Warn)
+				return 4;
+
+			if (logLevel == LogLevel.Error)
+				return 3;
+
+			return 2; // LogLevel.Fatal
+		}
+
+		private static void Populate(IDictionary<string, object> propertyBag, string key, object value)
+		{
+			if (string.IsNullOrEmpty(key))
+				return;
+
+			if (propertyBag.ContainsKey(key))
+			{
+				if (string.Equals(value.ToString(), propertyBag[key].ToString(), StringComparison.Ordinal))
+					return;
+
+				key += "_1";
+			}
```

Comment:
It looks like this appends a suffix to a key name to create a unique name. Should this be done in a loop?

Reply:
Yes, I guess there can be multiple properties, will change the `if` to a `while`